client/homebrew/editor/snippetbar/snippetbar.less

```diff
@@ -2,99 +2,116 @@
 @import (less) '././././themes/fonts/5e/fonts.less';
 
 .snippetBar {
-	@menuHeight      : 25px;
+	@menuHeight : 25px;
 	position         : relative;
-	height           : @menuHeight;
+	min-width        : 331px;
+	height           : auto;
 	color            : black;
 	background-color : #DDDDDD;
+	display: flex;
+	justify-content: space-between;
+	flex-wrap:wrap-reverse;
+
+	.snippets {
+		display         : flex;
+		justify-content : space-between;
+	}
 
 	.editors {
-		position        : absolute;
-		top             : 0px;
-		right           : 0px;
 		display         : flex;
 		justify-content : space-between;
-		height          : @menuHeight;
-		& > div {
-			width       : @menuHeight;
-			height      : @menuHeight;
-			line-height : @menuHeight;
-			text-align  : center;
-			cursor      : pointer;
-			&:hover,&.selected { background-color : #999999; }
-			&.text {
-				.tooltipLeft('Brew Editor');
-			}
-			&.style {
-				.tooltipLeft('Style Editor');
-			}
-			&.meta {
-				.tooltipLeft('Properties');
-			}
-			&.undo {
-				.tooltipLeft('Undo');
-				font-size : 0.75em;
-				color     : grey;
-				&.active { color : inherit; }
-			}
-			&.redo {
-				.tooltipLeft('Redo');
-				font-size : 0.75em;
-				color     : grey;
-				&.active { color : inherit; }
-			}
-			&.foldAll {
-				.tooltipLeft('Fold All');
-				font-size : 0.75em;
-				color     : inherit;
-			}
-			&.unfoldAll {
-				.tooltipLeft('Unfold All');
-				font-size : 0.75em;
-				color     : inherit;
+		>div {
+			display:flex;
+			justify-content: space-around;
+			flex:1;
+			border-left:1px solid;
+
+			&:first-child {
+				border-left: none;
 			}
-			&.history {
-				.tooltipLeft('History');
-				font-size : 0.75em;
-				color     : grey;
-				position  : relative;
-				&.active {
-					color : inherit;
+
+			& > div {
```

Comment:
I think here you need to have `position: relative;`, so that the hover tooltips display next to the button.  Couple of alternatives, though, if you want to consider them:

- They tooltips could be displayed next to their *group* rather than each button, in which case just set `position: relative;` one level up.
- They tooltips currently can go outside the container and are not visible if the snippetbar is wrapped.  I would actually create two sets of rules, one for tooltip below and one for tooltip on the left.  Set the "left" rule when not wrapped, and the "below" rule for when wrapped.  
- Or, just get rid of the fancy tooltips entirely and use the `title` attribute, which is more accessible, and has built-in collision detection so it will always be visible.  

Ultimately, CSS Anchor Positioning would fix this very very easily, but it is not available in FF yet.  If you wanted to go ahead with it anyway, you could do a `@supports` rule and have one set for Chrome, one for FF.  In fact, the entire system of snippet menus could use it for better results and less code.  Probably it's own PR though.

Reply:
restored position